Source/init.cpp

```diff
@@ -113,7 +113,7 @@ std::vector<std::string> GetMPQSearchPaths()
 		std::string message;
 		for (std::size_t i = 0; i < paths.size(); ++i) {
 			char prefix[32];
-			std::snprintf(prefix, sizeof(prefix), "\n%6u. '", static_cast<unsigned>(i + 1));
+			*fmt::format_to_n(prefix, sizeof(prefix) - 1, "\n{:6d}. '", static_cast<unsigned>(i + 1)).out = '\0';
 			message.append(prefix);
 			message.append(paths[i]);
 			message += '\'';

```

Comment:
This should be able to be simplified to

```cpp
message.append(fmt::format("{:6d} '{}'", i+1, paths[i]));
```

Reply:
That would allocate a string unnecessarily. We know the buffer is large enough, so replace `format_to_n` with `format_to`, making it easier to read.

Reply:
That code is only ever run if verbose logging is enabled, and only 2 or 3 times when the user starts the game or once when they change the language in the settings menu to something other than english. I was hopeful it'd be worth the tradeoff to make it simpler to read and not call `string::append` so much.

Reply:
Fair enough, I tend to write everything in the same efficient style so that I don't have to think too much about how often something is called.

If we had `absl::StrAppend`, this whole thing (and a number of other places) couldve been a single call, perhaps we could consider implementing something similar to it.

Reply:
Done